crates/noirc_evaluator/src/ssa_refactor/acir_gen/acir_ir/memory.rs

```diff
@@ -0,0 +1,145 @@
+use super::{acir_variable::AcirVar, errors::AcirGenError};
+
+#[derive(Debug, Default)]
+pub(crate) struct Array {
+    /// Elements in the array.
+    /// None represents the element not being set yet
+    ///
+    /// The size of this vector cannot be changed once set.
+    /// This follows the behavior of an Array as opposed to a Vector.
+    elements: Vec<Option<AcirVar>>,
+}
+
+impl Array {
+    pub(crate) fn size(&self) -> usize {
+        self.elements.len()
+    }
+
+    pub(crate) fn new(size: usize) -> Array {
+        Self { elements: vec![None; size] }
+    }
+
+    pub(crate) fn with_default(size: usize, default: AcirVar) -> Array {
+        Self { elements: vec![Some(default); size] }
+    }
+}
+
+#[derive(Debug, Default)]
+/// Memory used to represent Arrays
+pub(crate) struct Memory {
+    arrays: Vec<Array>,
+}
+
+impl Memory {
+    pub(crate) fn new() -> Self {
+        Self { arrays: Vec::new() }
+    }
+
+    /// Allocates an array of size `size`.
+    /// The elements in the array are not zero initialized
+    ///
+    /// TODO: Check if this method is needed, ie allocation without a default
```

Comment:
We can create an issue to define the behaviour when one allocates an array and then tries to access an element at any index. 

Currently, we throw, but perhaps we should zero initialize the values? It's probably safer to not zero intialize, we can punt this down the road until we get dynamic array so we have more information to go off of

Reply:
Issue: https://github.com/noir-lang/noir/issues/1458